### ISSUE TITLE
Image upload 500 error

### ISSUE THREAD

**Issue by fweng322** [mastodon/mastodon#28545]
### Steps to reproduce the problem

1. Upload an image when posting a tooth and web interface showed 500

...

### Expected behaviour

Image should be able to attach in a tooth

### Actual behaviour

Web interface showed "500"

### Detailed description

When uploading an image I got 500 error.

Similar to this issue:  https://github.com/mastodon/mastodon/issues/26961  but I'm in version 4.2.3.

I tried to set `RAILS_LOG_LEVEL=debug` in .env.production but didn't get more detail log in nginx access log other than:
```
[01/Jan/2024:19:19:27 +0800] "POST /api/v2/media HTTP/2.0" 500 77 "https://social.slat.org/deck/getting-started" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:121.0) Gecko/20100101 Firefox/121.0"
```

Not sure if it is related or not, but in nginx error log I kept getting "connection refused"

```
2024/01/01 19:22:00 [error] 631#631: *82551 connect() failed (111: Connection refused) while connecting to upstream, client: 210.236.231.149, server: social.slat.org, request: "POST /inbox HTTP/1.1", upstream: "http://127.0.0.1:3000/inbox", host: "social.slat.org"
```

Though I'm sure puma 6.3.1 is listening to port 3000.

```
root@mastodon:/var/log/nginx# ps axw | grep 3000
4111431 ?        Ssl    0:05 puma 6.3.1 (tcp://127.0.0.1:3000) [mastodon]
4111677 pts/1    S+     0:00 grep 3000
```

### Mastodon instance

social.slat.org

### Mastodon version

v4.2.3

### Technical details

If this is happening on your own Mastodon server, please fill out those:
- Ruby version: v3.2.2
- Node.js version: 16.20.2

**Comment by vmstan** [comment]
Did image uploads ever work?
Are you uploading to local storage or to an S3 object store?
How much memory is available on your system?

**Comment by fweng322** [comment]
> Did image uploads ever work?
> Are you uploading to local storage or to an S3 object store?
> How much memory is available on your system?

Thanks for your help.

I used local storage, but I just realized that my root partition was full, because backup partition wasn't mounted correctly so all the backup data was put into root partition and make it full.  After clearing it, image could be uploaded.

The mastodon folder including its storage is in a dedicate partition (/home/mastodon, in my system home is a dedicated partition, while backup is in another partition), which works without problem and the space is enough.  I wonder where the uploaded image goes (temporarily) when uploading images?  It seems to be put somewhere other than the mastodon folder.  Probably /tmp?
